### PR TITLE
Add a template runtime for Tensorflow

### PR DESCRIPTION
**Pull request by ipetr0v**
This PR copies an existing Echo Baremetal binary and makes it a template for the future TensorFlow Baremetal binary.

### CODE REVIEW

**Comment by tiziano88** [comment]
Please don't put more stuff under experimental, I'm trying to move stuff out of it

**Comment by ipetr0v** [comment]
> Please don't put more stuff under experimental, I'm trying to move stuff out of it

Moved to `tensorflow/...`

**Comment by tiziano88** [comment]
I would suggest just having the crates as top level, the nesting just makes it harder to find them, and I'm not sure it helps with anything in particular (or does it?).

**Comment by ipetr0v** [comment]
> I would suggest just having the crates as top level, the nesting just makes it harder to find them, and I'm not sure it helps with anything in particular (or does it?).

Done

File: oak_tensorflow_bin/layout.ld [hunk 1]
```diff
@@ -0,0 +1,111 @@
+/*
```
Comment (tiziano88) [inline oak_tensorflow_bin/layout.ld:1]: What's the plan to keep this in sync with the other binaries @andrisaar @conradgrobler  ?

**Review by tiziano88 — APPROVED** [review]

File: oak_tensorflow_bin/layout.ld (hunk 1 above)
Comment (conradgrobler) [reply oak_tensorflow_bin/layout.ld:1]: We don't yet have a good plan for the linker scripts. I vaguely remember that one linker script can reference another, so we might be able to move a lot of it to a central script.

We should reuse the asm between the different binaries and do general deduplication where possible before creating any new binaries that are based on the restricted kernel.

Comment (tiziano88) [reply oak_tensorflow_bin/layout.ld:1]: Perhaps we could use some symlinks for the time being, instead of maintaining several copies in our repo?

**Comment by ipetr0v** [comment]
Ref https://github.com/project-oak/oak/issues/3297

**Pull request merged**